File: frontend/lib/category/models/category_stats.dart
/// A class representing the statistics for categories.
class CategoryStats {
  /// The number of transactions matching to each category.
  final Map<String, int> categoryCount;

  CategoryStats({required this.categoryCount});

  factory CategoryStats.fromJson(Map<String, dynamic> json) {
    return CategoryStats(categoryCount: Map<String, int>.from(json['categoryCount'] as Map));
  }
}

File: backend/src/category/model/api/category.stats.ts
import { Base } from "@backend/core/model/base";

/** Information regarding the categories and their use for a user  */
export class CategoryStats extends Base {
  /** The number of transactions matching to each category */
  categoryCount: { [name: string]: number };

  constructor(categoryCount: CategoryStats["categoryCount"]) {
    super();
    this.categoryCount = categoryCount;
  }
}
